### PR TITLE
Sync clang-format with OpenLoco

### PR DESCRIPTION
**Pull request by janisozaur**
Just the file, without reformatting the sources yet.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,26 +1,107 @@
...
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeComma
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     0
+CommentPragmas:  '^ IWYU pragma:'
```
Comment (Broxzier) [inline .clang-format]: Maybe we could use this to prevent the copyright comment from being formatted.

File: .clang-format [hunk 2]
```diff
@@ -1,26 +1,107 @@
...
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: false
+ConstructorInitializerIndentWidth: 4
+ContinuationIndentWidth: 4
+Cpp11BracedListStyle: false
```
Comment (Broxzier) [inline .clang-format]: I think it's nice to set this to true. Example taken from llvm.org: `new int[3]{1, 2, 3};` becomes `new int[3]{ 1, 2, 3 };`

File: .clang-format [hunk 3]
```diff
@@ -1,26 +1,107 @@
-﻿---
-BasedOnStyle: LLVM
-AccessModifierOffset: '-4'
+---
+Language:        Cpp
+# BasedOnStyle:  WebKit
```
[the remark this answers is not in the document]
Comment (janisozaur) [reply .clang-format]: I expect it is, I think @marijnvdwerf simply exploded the WebKit style and adapted it to project needs.

File: .clang-format [hunk 4]
```diff
@@ -1,26 +1,107 @@
...
+SpaceBeforeParens: ControlStatements
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Cpp11
+TabWidth:        8
```
[the remark this answers is not in the document]
Comment (janisozaur) [reply .clang-format]: This is `TabWidth`, it shouldn't matter as we don't use tabs.

File: .clang-format [hunk 5]
```diff
@@ -1,26 +1,107 @@
...
-SortIncludes: 'true'
-Standard: Cpp11
-TabWidth: '4'
-
+  - Regex:           '^(<|"(gtest|gmock|isl|json)/)'
+    Priority:        3
+  - Regex:           '.*'
+    Priority:        1
+IncludeIsMainRegex: '(Test)?$'
+IndentCaseLabels: true
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply .clang-format]: Personally, I'd be happy to start indenting switch block contents, too. Especially with cases that contain blocks (braces), it's much harder to see whether they were terminated properly when we do not indent.

File: .clang-format [hunk 6]
```diff
@@ -1,26 +1,107 @@
...
+IndentWidth:     4
+IndentWrappedFunctionNames: false
+JavaScriptQuotes: Leave
+JavaScriptWrapImports: true
+KeepEmptyLinesAtTheStartOfBlocks: true
+MacroBlockBegin: ''
+MacroBlockEnd:   ''
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: All
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply .clang-format]: Hmm, personally, I prefer not to indent within namespaces _at all_, but it's not a strong preference.

File: .clang-format [hunk 7]
```diff
@@ -1,26 +1,107 @@
...
+ObjCSpaceAfterProperty: true
+ObjCSpaceBeforeProtocolList: true
+PenaltyBreakAssignment: 2
+PenaltyBreakBeforeFirstCallParameter: 19
+PenaltyBreakComment: 300
+PenaltyBreakFirstLessLess: 120
+PenaltyBreakString: 1000
+PenaltyExcessCharacter: 1000000
+PenaltyReturnTypeOnItsOwnLine: 60
+PointerAlignment: Left
```
Comment (Gymnasiast) [inline .clang-format]: :'(

I don't like this, really. But then again, there is something to be said for keeping OpenLoco and OpenRCT2's rules the same.

Comment (AaronVanGeffen) [reply .clang-format]: Agreed with @IntelOrca. Personally, I don't feel too strongly for either left or right, but the centre alignment looks too much like multiplication to me.

Comment (Gymnasiast) [reply .clang-format]: There is still the point whether we adopt OpenLoco's rules or not. But I doubt many people are against that.

Comment (AaronVanGeffen) [reply .clang-format]: @Broxzier has commented with a few valid points in this PR. Other than that, I'm all for adopting OpenLoco's clang-format style.

File: .clang-format (hunk 3 above)
[the remark this answers is not in the document]
Comment (marijnvdwerf) [reply .clang-format]: Get's generated automatically when you do `clang-format -style=webkit -dump-config`

File: .clang-format [hunk 8]
```diff
@@ -1,26 +1,107 @@
...
-Cpp11BracedListStyle: 'false'
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlines: Right
+AlignOperands:   false
+AlignTrailingComments: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: All
```
[the remark this answers is not in the document]
Comment (Gymnasiast) [reply .clang-format]: Indeed, either None or not touching it at all.

File: .clang-format (hunk 2 above)
Comment (Gymnasiast) [reply .clang-format]: The latter appears to be more in line with how we do it currently.

File: .clang-format [hunk 9]
```diff
@@ -1,26 +1,107 @@
...
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeComma
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     0
```
Comment (Broxzier) [inline .clang-format]: 128.

**Pull request merged**